tuning/sft_trainer.py

```diff
@@ -613,19 +648,27 @@ def main(**kwargs):  # pylint: disable=unused-argument
         sys.exit(INTERNAL_ERROR_EXIT_CODE)
 
     # save model
-    if training_args.save_model_dir:
-        try:
+    try:
+        if training_args.save_model_dir:
             save(
                 path=training_args.save_model_dir,
                 trainer=trainer,
                 log_level=training_args.log_level,
             )
-        except Exception as e:  # pylint: disable=broad-except
-            logger.error(traceback.format_exc())
-            write_termination_log(
-                f"Failed to save model to {training_args.save_model_dir}: {e}"
+        else:
+            # if granite with llama arch, remove lm_head in last checkpoint
+            save(
+                path=get_highest_checkpoint(training_args.output_dir),
+                trainer=trainer,
+                log_level=training_args.log_level,
```

Comment:
This is currently removing lm_head from the model that is being saved at save_model_dir OR the last checkpoint. A few questions...
- Do we need to remove from last checkpoint? This is for the use case if someone doesn't specify save_model_dir and retains current behavior but it is a little weird for the final checkpoint to be different than the rest of the checkpoints.
- Do we want to remove from each checkpoint? This would require loading up each of the checkpoints again and resaving it, unless SFTTrainer.trainer has access to the checkpoints still but AFAIK it only has access to the last checkpoint